R/bundlePython.R

```diff
@@ -0,0 +1,138 @@
+# Create anonymous function that we can later call to get all needed python
+# metdata for the manifest
+pythonConfigurator <- function(python,
+                               forceGenerate = FALSE,
+                               condaMode = FALSE) {
+
+  if (is.null(python)) {
+    return(NULL)
+  }
+
+  force(condaMode)
+  force(forceGenerate)
+
+  function(appDir) {
+    withCallingHandlers(
+      inferPythonEnv(
+        appDir,
+        python = python,
+        condaMode = condaMode,
+        forceGenerate = forceGenerate
+      ),
+      error = function(err) {
+        cli::cli_abort(
+          "Failed to detect python environment",
+          parent = err
+        )
+      }
+    )
+  }
+}
+
+appUsesPython <- function(quartoInfo) {
+  if (is.null(quartoInfo)) {
+    # No R-based, non-Quarto content uses Python by default.
+    # Looking for Python chunks in Rmd needs to happen separately.
+    FALSE
+  } else {
+    # Python is a direct consequence of the "jupyter" engine; not "knitr" or "markdown".
+    "jupyter" %in% quartoInfo[["engines"]]
+  }
+}
+
+# python is enabled on Connect and posit.cloud, but not on Shinyapps
+getPythonForTarget <- function(path, accountDetails) {
+  targetIsShinyapps <- isShinyappsServer(accountDetails$server)
+  pythonEnabled <- getOption("rsconnect.python.enabled", !targetIsShinyapps)
+  if (pythonEnabled) {
+    getPython(path)
+  } else {
+    NULL
+  }
+}
+
+getPython <- function(path = NULL) {
+  if (!is.null(path)) {
+    return(path.expand(path))
+  }
+
+  path <- Sys.getenv("RETICULATE_PYTHON")
+  if (path != "") {
+    return(path.expand(path))
+  }
+
+  path <- Sys.getenv("RETICULATE_PYTHON_FALLBACK")
+  if (path != "") {
+    return(path.expand(path))
+  }
+
+  NULL
+}
+
+inferPythonEnv <- function(workdir,
+                           python = getPython(),
+                           condaMode = FALSE,
+                           forceGenerate = FALSE) {
+  # run the python introspection script
+  env_py <- system.file("resources/environment.py", package = "rsconnect")
+  args <- c(
+    shQuote(env_py),
+    if (condaMode) "-c",
+    if (forceGenerate) "-f",
+    shQuote(workdir)
+  )
+
+  # First check for reticulate. Then see if python is loaded in reticulate
+  # space, verify anaconda presence, and verify that the user hasn't specified
+  # that they don't want their conda environment captured.
+  hasConda <- is_installed("reticulate") &&
```

Comment:
@aronatkins so `condaMode = TRUE` used to imply that you _didn't_ want to capture your conda environment?

Reply:
I'm not really sure what interaction this was trying to capture. We always call `resources/environment.py`, just with different arguments and a different interpreter.

```bash
# hasConda=TRUE, condaMode=FALSE
conda run -p <PYTHON> resources/environment.py 
# hasConda=TRUE, condaMode=TRUE
# IMPOSSIBLE
# hasConda=FALSE, condaMode=FALSE
<PYTHON> resources/environment.py
# hasConda=FALSE, condaMode=TRUE
<PYTHON> resources/environment.py -c
```

Since `condaMode` is never enabled, it looks like we can strip all support (including the `-c` argument to `resources/environment.py`). The `hasConda` logic feels like it still needs to remain.